libs/blocks/bulk/bulk.js

```diff
@@ -0,0 +1,341 @@
+
+  const executeBulk = async (resume) => {
+    // reset the result area
+    setResult(null);
+    setCompletion(null);
+    setValid(null);
+
+    // validate the user
+    const authorizedValue = await userIsAuthorized();
+    setAuthorized(authorizedValue);
+    if (!authorizedValue) {
+      setValid(false);
+      return;
+    }
+
+    // validate the number of urls
+    const { urls } = getStoredOperation();
+    const urlNumberValue = urls.length;
+    setUrlNumber(urlNumberValue);
+    if (urlNumberValue < 1 || urlNumberValue > URLS_ENTRY_LIMIT) {
+      setValid(false);
+      return;
+    }
+
+    // perform the action
+    setValid(true);
+    const operation = getStoredOperation().name;
+    const results = await executeActions(resume, setResult);
+    const completionValue = getCompletion(results);
+    setCompletion(completionValue);
+
+    // log the actions on the server
+    await sendReport(results, operation);
+  };
+
+  const onSelectChange = () => {
+    setSelectedAction(actionElt.current.value);
+  };
+
+  const hideModal = () => {
+    resumeModal.current.classList.remove('displayed');
+  };
+
+  const submit = async () => {
+    setBulkTriggered(true);
+    const operation = actionElt.current.value;
+    setSubmittedAction(operation);
+    storeOperation(operation);
+    const urls = getUrls(urlsElt);
+    storeUrls(urls);
+    await executeBulk();
+  };
+
+  const resume = async () => {
+    setBulkTriggered(true);
+    hideModal();
+    await executeBulk(true);
+  };
+
+  return html`
+  <div class="bulk-container">
+    <div class="bulk-input" ref="${bulkInputElt}">
+      <div class="bulk-header">
+        <div class="tool-header">
+          <h1>Bulk Publishing</div>
+        </div>
+        <${User} user=${user} />
+      </div>
+      <div class="bulk-controls">
+        <div class="bulk-urls-title">Add URLS</div>
+        <div class="bulk-action">
+          <${SelectBtn}
+            actionElt=${actionElt}
+            onSelectChange=${onSelectChange}
+            storedOperationName=${storedOperation.name} />
+          <${SubmitBtn}
+            submit=${submit}
+            selectedAction=${selectedAction} />
+        </div>
+      </div>
+      <${UrlInput} urlsElt=${urlsElt} />
+    </div>
+    <${ErrorMessage}
+      valid=${valid}
+      authorized=${authorized}
+      urlNumber=${urlNumber} />
+    <${Status}
+      valid=${valid}
+      urlNumber=${urlNumber}
+      bulkTriggered=${bulkTriggered}
+      submittedAction=${submittedAction}
+      result=${result}
+      resultsElt=${resultsElt}
+      completion=${completion} />
+    <${ResumeModal}
+      displayResumeDialog=${displayResumeDialog}
+      resumeModal=${resumeModal}
+      resume=${resume}
+      hideModal=${hideModal} />
+  </div>
+  `;
+}
+
+export default async function init(el) {
+  if (IMS_SIGN_IN_ENABLED) {
+    const signedIn = await signIn(IMS_SIGN_IN_ENABLED);
```

Comment:
There is no need to pass in an argument here.

Reply:
@honstar Good catch! I addressed in with https://github.com/adobecom/milo/pull/559/commits/80e5403a54c434c52a6e93baa6ba71b80ea23c4a